Tests/UserServiceTests/Validators/UserEmailPasswordValidatorTests.cs

```diff
@@ -0,0 +1,402 @@
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameError);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void WrongBirthday()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Now.AddDays(1)
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.BirthdayYoung);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissBirthday()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = "email@yandex.ru",
+                Password = "123",
+                FirstName = "Leo",
+                LastName = "Kor"
+            };
+            
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.BirthdayEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissEmail()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Password = "123",
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.EmailEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissPassword()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.EmptyPassword);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissFirstName()
```

Comment:
Haven't you already checked it in EmptyFirstName? And I guess then we should choose either `Empty` or `Miss`

Reply:
Yes, you are right. Now empty test where field is empty string and miss test where this field missed. Fixed